animation_nodes/nodes/vector/bl_vector_fractal.py

```diff
@@ -0,0 +1,119 @@
+    ("VORONOI_F3", "Voronoi F3", "", "", 4),
+    ("VORONOI_F4", "Voronoi F4", "", "", 5),
+    ("VORONOI_F2F1", "Voronoi F2F1", "", "", 6),
+    ("VORONOI_CRACKLE", "Voronoi Crackle", "", "", 7),
+    ("CELLNOISE", "Cell Noise", "", "", 8)
+]
+
+class BLVectorFractalNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_BLVectorFractal"
+    bl_label = "BL Vector Fractal"
+    bl_width_default = 160
+
+    __annotations__ = {}
+
+    __annotations__["noisemode"] = EnumProperty(name = "Type ", default = "FRACTAL",
+        items = noiseModeItems, update = AnimationNode.refresh)
+
+    __annotations__["noisebasis"] = EnumProperty(name = "Basis ", default = "PERLIN_ORIGINAL",
+        items = vectorFractalModeItems, update = AnimationNode.refresh)
+
+    useVectorList: VectorizedSocket.newProperty()    
+
+    def create(self):
+        self.newInput(VectorizedSocket("Vector", "useVectorList",
+            ("Vector", "v"), ("Vectors", "vectorsIn")))       
+        self.newInput("Float", "H", "h")
+        self.newInput("Float", "Lacunarity", "lacunarity")
+        self.newInput("Integer", "Octaves", "octaves", value = 2, minValue = 1)
+        if self.noisemode == "HETERO_TERRAIN":
+            self.newInput("Float", "Offset", "offset")
+        if self.noisemode == "RIDGED_MULTI_FRACTAL" or self.noisemode == "HYBRID_MULTI_FRACTAL" :
+            self.newInput("Float", "Offset", "offset")
+            self.newInput("Float", "Gain", "gain")   
+
+        self.newOutput(VectorizedSocket("Float", ["useVectorList"],
+            ("Noise", "noise_out"), ("Noise", "noises_out")))
+    
+    def draw(self, layout):
+        layout.prop(self, "noisemode")
+        layout.prop(self, "noisebasis")
+
+    def getExecutionFunctionName(self):
+        if self.useVectorList:
+            if self.noisemode == "FRACTAL":
+                return "execute_fractal"
+            elif self.noisemode == "MULTI_FRACTAL":
+                return "execute_multifractal"
+            elif self.noisemode == "HETERO_TERRAIN":
+                return "execute_heteroterrain"
+            elif self.noisemode == "RIDGED_MULTI_FRACTAL":
+                return "execute_rigidmultifractal"
+            elif self.noisemode == "HYBRID_MULTI_FRACTAL":
+                return "execute_hybridmultifractal"                
+        else:
+            if self.noisemode == "FRACTAL":
+                return "execute_fractal_s"
+            elif self.noisemode == "MULTI_FRACTAL":
+                return "execute_multifractal_s"
+            elif self.noisemode == "HETERO_TERRAIN":
+                return "execute_heteroterrain_s"
+            elif self.noisemode == "RIDGED_MULTI_FRACTAL":
+                return "execute_rigidmultifractal_s"
+            elif self.noisemode == "HYBRID_MULTI_FRACTAL":
+                return "execute_hybridmultifractal_s"
+
+    def execute_fractal(self, vectorsIn, h, lacunarity ,octaves):
+        noisebasis = self.noisebasis
```

Comment:
Instead of checking for `self.useVectorList` at line 63, here, you can write something like this,
```
    def execute_Fractal(self, vectors, h, lacunarity ,octaves):
        if not self.useVectorList: vectors = [vectors]
        noisebasis = self.noisebasis
```
Then no need for separate functions for single value.

Reply:
I didn't know that. great technique. will apply these changes